service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -301,11 +304,22 @@ public void deleteWorkspace(Workspace workspace, AuthenticatedUserRequest userRe
    *
    * @param workspaceUuid workspace of interest
    * @param propertyKeys list of keys in properties
+   * @param userEmail user email fetched from SAM based on the {@link AuthenticatedUserRequest}
+   * @param subjectId subject id fetched from SAM based on the {@link AuthenticatedUserRequest}
    */
-  public void deleteWorkspaceProperties(UUID workspaceUuid, List<String> propertyKeys) {
+  public void deleteWorkspaceProperties(
+      UUID workspaceUuid, List<String> propertyKeys, String userEmail, String subjectId) {
     workspaceDao.deleteWorkspaceProperties(workspaceUuid, propertyKeys);
     workspaceActivityLogDao.writeActivity(
-        workspaceUuid, new DbWorkspaceActivityLog().operationType(OperationType.UPDATE));
+        workspaceUuid, getDbWorkspaceActivityLog(OperationType.DELETE, userEmail, subjectId));
+  }
+
+  private DbWorkspaceActivityLog getDbWorkspaceActivityLog(
```

Comment:
You have several classes with this helper method, can you move to someplace common?